services/navigation/builtin/builtin.go

```diff
@@ -495,42 +509,86 @@ func (svc *builtIn) Close(ctx context.Context) error {
 	return svc.store.Close(ctx)
 }
 
+func (svc *builtIn) moveOnGlobeSync(ctx context.Context, wp navigation.Waypoint, extra map[string]interface{}) error {
+	req := motion.MoveOnGlobeReq{
+		ComponentName:      svc.base.Name(),
+		Destination:        wp.ToPoint(),
+		Heading:            math.NaN(),
+		MovementSensorName: svc.movementSensor.Name(),
+		Obstacles:          svc.obstacles,
+		MotionCfg:          svc.motionCfg,
+		Extra:              extra,
+	}
+	var stopWG sync.WaitGroup
+	defer stopWG.Wait()
+	cancelCtx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+	rawExecutionID, err := svc.motionService.MoveOnGlobeNew(cancelCtx, req)
+	if err != nil {
+		return err
+	}
+
+	executionID, err := uuid.Parse(rawExecutionID)
+	if err != nil {
+		return err
+	}
+	executionWaypoint := executionWaypoint{executionID: executionID, waypoint: wp}
+	if old := svc.activeExecutionWaypoint.Swap(executionWaypoint); old != nil && old != emptyExecutionWaypoint {
+		msg := "unexpected race condition in moveOnGlobeSync, expected " +
+			"replaced waypoint & execution id to be nil or %#v; instead was %s"
+		svc.logger.Errorf(msg, emptyExecutionWaypoint, old)
+	}
+	stopWG.Add(1)
+	// call StopPlan upon exiting moveOnGlobeSync
+	// is a NoOp if execution has already terminted
+	utils.ManagedGo(func() {
+		<-cancelCtx.Done()
```

Comment:
I dont think this has to be its own thing in a goroutine and that we can just call stop when we need to stop

Reply:
You are right. I've refactored it to be a defer